### PR TITLE
feat(examples): add offline tool-call eval for agent SFT checkpoints

### PR DESCRIPTION
**Pull request by khazic**
# What does this PR do ?

Adds an **offline tool-call accuracy evaluator** for agent SFT checkpoints — the missing entry point the recipe already points users to.

`ToolCallAccuracyEvaluator` exists and is framework-agnostic, and the training recipe's warning literally says *"run the evaluator offline from a checkpoint"* — but there was no script to do it, and the FSDP2 recipe **skips in-loop generation eval by default**. So a trained agent model's tool-call accuracy (the signal `val_loss` is blind to: wrong tool names, malformed argument JSON) was effectively unmeasurable through any supported path.

`examples/llm_finetune/agent/evaluate_tool_call.py` loads a finished checkpoint or any HF model, runs the evaluator over a held-out function-calling set, and prints the `tool_call/*` metrics.

```bash
# Consolidated (HF-compatible) checkpoint + held-out jsonl
python examples/llm_finetune/agent/evaluate_tool_call.py \
  --checkpoint-path /path/to/checkpoint/epoch_X_step_Y \
  --data-path /path/to/heldout_function_calling.jsonl

# DCP / LoRA checkpoint: also pass the base model
python examples/llm_finetune/agent/evaluate_tool_call.py \
  --checkpoint-path /path/to/ckpt --base-model-path Qwen/Qwen2.5-3B \
  --data-path /path/to/heldout.jsonl

# Baseline an untrained model from the Hub
python examples/llm_finetune/agent/evaluate_tool_call.py \
  --base-model-path Qwen/Qwen2.5-3B --dataset-name llamafactory/glaive_toolcall_en --split test
```

# Changelog

- Add `examples/llm_finetune/agent/evaluate_tool_call.py`: offline tool-call accuracy eval. Checkpoint loading mirrors `examples/vlm_generate/generate.py` (consolidated safetensors / DCP / LoRA, or a plain HF id). Eval source is a local jsonl (`--data-path`) or a Hub split (`--dataset-name`/`--split`).
- Usage docstring recommends a genuinely held-out split and notes `train[:N]` only smoke-tests (addresses the "eval on `train[:N]`" gap).

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [ ] Tests: example script (like the sibling `generate.py`/`finetune.py`, no unit test); the underlying `ToolCallAccuracyEvaluator` is unit-tested.
- [x] Docs: self-documenting usage in the script header.

# Additional Information

- Single-process / single-GPU offline eval — no FSDP, so none of the in-loop generation-under-FSDP2 OOM/skip constraints apply.
- `ruff` clean; imports resolve and the script compiles. End-to-end run needs a GPU + checkpoint (not runnable in CPU CI).
- DCO signed-off.

### CODE REVIEW

**Review by HuiyingLi — APPROVED** [review]
Script is intended for single gpu. Worth a comment noting large models are not supported but non-blocking. Thank you!
